### PR TITLE
Backend of app is done

### PR DESCRIPTION
**Pull request by letit6E**
Application backend implemented, streamlit added

### CODE REVIEW

**Review by SpirinEgor — CHANGES_REQUESTED** [review]
Avoid global code for both the Streamlit interface and pipeline initialization.

File: main.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import streamlit as st
+
+from main_summarizer import Summarizer
+from syntax_summarizer import get_text
+
+if __name__ == '__main__':
+	cnn_summarizer = Summarizer()
```
Comment (SpirinEgor) [inline main.py:7]: This is still global code, you just put it under the `if` statement  🤷‍♂️
In this way, code is less reusable

File: main_summarizer.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	MAX_TOKEN_COUNT = 1024
+	TEXT_LENGTH_INTERVAL = 100
+
+	def __init__(self):
+		self.pipe = pipeline("summarization", model="facebook/bart-large-cnn")
+
+	def summarize(self, text, min_size):
+		trimmed_text = text
+
+		while len(self.pipe.tokenizer(trimmed_text).tokens()) > self.MAX_TOKEN_COUNT:
```
Comment (SpirinEgor) [inline main_summarizer.py:14]: Tokenization is a deterministic process, you should call it once.

**Review by SpirinEgor — APPROVED** [review]
A little bit strange solution, it is definitely not in a pythonic way and not optimized at all. But it is working 👍

File: main_summarizer.py (hunk 2 above)
Comment (letit6E) [reply main_summarizer.py:14]: But I'm changing the variable trimmed_text in a loop, what do you mean by "deterministic" when the function is called with different inputs?

> Tokenization is a deterministic process, you should call it once.

Comment (SpirinEgor) [reply main_summarizer.py:14]: You can do something like that:
```python
tokens = tokenizer(text).tokens()
tokens = tokens[:self.MAX_TOKEN_COUNT]
```

**Pull request merged**